go/cagra/index_params.go

```diff
@@ -0,0 +1,170 @@
+package cagra
+
+// #include <cuda_runtime_api.h>
+// #include <cuvs/core/c_api.h>
+// #include <cuvs/distance/pairwise_distance.h>
+// #include <cuvs/neighbors/brute_force.h>
+// #include <cuvs/neighbors/ivf_flat.h>
+// #include <cuvs/neighbors/cagra.h>
+// #include <cuvs/neighbors/ivf_pq.h>
+import "C"
+import (
+	"errors"
+	"unsafe"
+
+	cuvs "github.com/rapidsai/cuvs/go"
+)
+
+type IndexParams struct {
+	params C.cuvsCagraIndexParams_t
+}
+
+type CompressionParams struct {
+	params C.cuvsCagraCompressionParams_t
+}
+
+type BuildAlgo int
+
+const (
+	IvfPq BuildAlgo = iota
+	NnDescent
+	AutoSelect
+)
+
+var cBuildAlgos = map[BuildAlgo]int{
+	IvfPq:      C.IVF_PQ,
+	NnDescent:  C.NN_DESCENT,
+	AutoSelect: C.AUTO_SELECT,
+}
+
+func CreateCompressionParams() (*CompressionParams, error) {
```

Comment:
Can we please document all of these functions? Does Go have a common convention or tooling for generating standardized documentation?